### PR TITLE
Add examples in ci

### PR DESCRIPTION
**Pull request by mwestphal**
- Fix an issue with PUBLIC link in plugins
- Fix a potential segfault with an invalid file
- Add examples
- Use examples instead of test_sdk to validate install in CI
- Build and validate plugins in CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -193,25 +194,36 @@ jobs:
...
+      with:
+        path: ${{github.workspace}}\build_examples\render-image\Release\
+
+    - name: Install Mesa Windows render-interact
+      if: matrix.os == 'windows-latest'
+      uses: ./source/.github/actions/mesa-install
+      with:
+        path: ${{github.workspace}}\build_examples\render-interact\Release\
+
+    # older VTK version create different renderings so the are not tested
```
Comment (mwestphal) [inline .github/workflows/ci.yml:215]: they

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -101,13 +101,14 @@ jobs:
           matrix.vtk_version == 'v9.1.0' )
       working-directory: ${{github.workspace}}/dependencies/
       shell: bash
-      run: echo "LD_LIBRARY_PATH=$(pwd)/install/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
+      run: |
+        echo "LD_LIBRARY_PATH=$(pwd)/install/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
```
Comment (mwestphal) [inline .github/workflows/ci.yml:105]: single line

**Comment by mwestphal** [comment]
Coverage change do not make sense to me, lets just ignore it ?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -193,25 +194,36 @@ jobs:
...
-    # TODO when VTK is built as a shared library:
-    # - Enable on all OS
-    # - Add -Werror=dev CMake option
-    # - Run ctest
-    - name: Validate plugin example
-      if: matrix.os == 'ubuntu-latest'
+        cmake -Werror=dev -Werror=deprecated --warn-uninitialized -Df3d_DIR:PATH=$(pwd)/install/lib/cmake/f3d -DBUILD_TESTING=ON -B build_examples source/examples/libf3d
+        cmake --build build_examples --config Release
+
+    - name: Install Mesa Windows render-image
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply .github/workflows/ci.yml:203]: I agree, but to do that I have to add cmake logic in the example, which increase their complexity for CI reasons, which I do not like at all.

Another solution is to support wildcards in the install-mesa-action, would be nice but i dont know if its hard to implement in powershell.

**Review by Meakk — APPROVED** [review]

**Review by Meakk — APPROVED** [review]
lgtm when CI is green

**Pull request merged**